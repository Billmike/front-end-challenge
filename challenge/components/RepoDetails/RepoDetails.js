import React from 'react';
import { View, Text, Dimensions, Image , TouchableOpacity} from 'react-native';

const styles = {
  dot: {
    borderRadius: 100,
    padding: 5,
    height: 10,
    width: 10,
    position: 'absolute',
    alignSelf: 'flex-end',
    marginBottom: 5,
    marginLeft: 'auto',
    top: 20,
    left: Dimensions.get('window').width - 90,
  }
}

const RepoDetails = (props) => {
  return props.repository.map((repos, i) => {
    return (
      <View style={{ display: 'flex', flex: 1, borderBottomColor: '#D3D3D3', borderBottomWidth: 1, marginBottom: 20, marginTop: 10 }} key={i}>
        <Text style={{ marginBottom: 15, fontSize: 16 }}>Repo Name: {repos.name}</Text>
        <View style={{ display: 'flex', flexDirection: 'row' }}>
          <Image source={{ uri: repos.owner.avatarUrl }} style={{width: 66, height: 58}} />
          <Text style={{ marginLeft: 10, fontSize: 16 }}>Owner: {repos.owner.login}</Text>
          <TouchableOpacity style={{ marginLeft: 'auto', marginRight: 10, marginTop: 5 }} onPress={() => props.onRemoveRepo(i)}>
            <Text>Remove Repo</Text>
          </TouchableOpacity>
        </View>
        <Text style={{ marginTop: 15 }}>Description: {repos.description}</Text>
        <Text style={{ marginTop: 15 }}>Pushed At: {repos.pushedAt}</Text>
        <Text style={{ marginTop: 10 }}>Stars(Star Gazers): {repos.stargazers.totalCount}</Text>
        <Text style={{ marginTop: 10 }}>Fork Count: {repos.forkCount}</Text>
        <Text style={{ marginTop: 15, marginBottom: 10 }}>Primary Language: {repos.primaryLanguage.name}</Text>
        <View style={[styles.dot, {backgroundColor: repos.primaryLanguage.color},]} />
      </View>
    )
  })
}

export default RepoDetails;
