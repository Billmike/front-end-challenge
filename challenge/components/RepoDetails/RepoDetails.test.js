import React from 'react';
import RepoDetails from './';

import renderer from 'react-test-renderer';

const props = [{
  name: 'James',
  description: 'A short repo',
  owner: {
    login: 'lambda',
    avatarUrl: 'https://dummyavatarurl'
  },
  pushedAt: '26-dec-2018',
  forkCount: 89,
  primaryLanguage: {
    color: '#FFF'
  },
  stargazers: {
    totalCount: 0
  }
}]
it('renders correctly', () => {
  const rendered = renderer.create(<RepoDetails repository={props} />).toJSON();
  expect(rendered).toMatchSnapshot()
})