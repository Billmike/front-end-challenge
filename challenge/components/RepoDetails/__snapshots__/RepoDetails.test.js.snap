// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D3D3D3",
      "borderBottomWidth": 1,
      "display": "flex",
      "flex": 1,
      "marginBottom": 20,
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginBottom": 15,
      }
    }
  >
    Repo Name: 
    James
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://dummyavatarurl",
        }
      }
      style={
        Object {
          "height": 58,
          "width": 66,
        }
      }
    />
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Owner: 
      lambda
    </Text>
    <Text
      style={
        Object {
          "left": 450,
          "position": "absolute",
          "top": 689,
        }
      }
    >
      Stars(Star Gazers): 
      0
    </Text>
    <Text
      style={
        Object {
          "left": 450,
          "position": "absolute",
          "top": 709,
        }
      }
    >
      Fork Count: 
      89
    </Text>
  </View>
  <Text
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    Description: 
    A short repo
  </Text>
  <Text
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    Pushed At: 
    26-dec-2018
  </Text>
  <Text
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 15,
      }
    }
  >
    Primary Language: 
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "borderRadius": 100,
          "height": 10,
          "left": 660,
          "marginBottom": 5,
          "marginLeft": "auto",
          "padding": 5,
          "position": "absolute",
          "top": 20,
          "width": 10,
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  />
</View>
`;
